drivers/spi/spi_ll_stm32.c

```diff
@@ -733,6 +733,13 @@ static int transceive_dma(const struct device *dev,
 		return -ENOTSUP;
 	}
 
+#ifdef CONFIG_SOC_SERIES_STM32H7X
+	if ((tx_bufs != NULL && !spi_buf_set_in_nocache(tx_bufs)) ||
+		(rx_bufs != NULL && !spi_buf_set_in_nocache(rx_bufs))) {
+		return -EFAULT;
```

Comment:
I wouldn't say exactly not compatible. In these 2 cases, if STM32H7 SPI is used and DMA is enabled, these functions will silently fail, so I still think it's better to return an error code.

Reply:
> Of course, I prefer returning an error code instead of silently failing. But I was thinking if we could aim for something like: ...

Hi, I am currently facing the task of installing BLE on the STM32H747 via hci_spi. As long as I use `CONFIG_SPI_STM32=y` or `CONFIG_SPI_STM32_INTERRUPT=y`, it works fine. But with `CONFIG_SPI_STM32_DMA=y` I get `error -14` (EFAULT), from exactly this line.

Am I right in assuming that the hci_spi driver - at least on the STM32H7 - currently has no DMA support? Would the buffers in drivers/bluetooth/hci/spi.c have to be provided with `__attribute__((__section__(".nocache")))` on the STM32H7?

Any suggestions on how I should proceed to use hci_spi with DMA?
Thanks for the help!

Reply:
> Am I right in assuming that the hci_spi driver - at least on the STM32H7 - currently has no DMA support? Would the buffers in drivers/bluetooth/hci/spi.c have to be provided with __attribute__((__section__(".nocache"))) on the STM32H7?

Hi, if you want to use SPI in H7 with DMA enabled, the buffers provided to the SPI transceive function must be in a nocache memory region. The preferred way to do this explained in this document: `${ZEPHYR_HOME}/doc/services/mem_mgmt/index.rst`

In summary, you can declare a memory region as nocache in devicetree, then use `__attribute__((section(...)))` to place the required buffer in such region.

Reply:
@Benni77 Yes, it makes sense. We need to be able to return a meaningful error in such case.